core/src/main/java/feign/codec/StreamDecoder.java

```diff
@@ -0,0 +1,21 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feign.codec;
+
+/**
+ * Custom decoder responsible to close the response stream itself.
+ * Used in java8 streaming api.
```

Comment:
I think this is misleading. This has very little to do with the java 8 streams, generically speaking.  for example, normal InputStream is the same, or Iterator<Foo> or anything that does incremental parsing, right?

I don't like the type-code here as it is more about the response type than the type of the decoder. For example, the same decoder will have non streaming types dispatched to it as well.